librubyfmt/src/format.rs

```diff
@@ -1701,14 +1642,10 @@ pub fn format_heredoc_string_literal(
         Box::new(|ps| {
             let heredoc_type = (hd.1).0;
             let heredoc_symbol = (hd.1).1;
-            ps.emit_heredoc_start(heredoc_type.clone(), heredoc_symbol.clone());
+            let kind = HeredocKind::from_string(&heredoc_type);
+            ps.emit_heredoc_start(heredoc_type, heredoc_symbol.clone(), kind);
```

Comment:
if we're modifying emit_heredoc_start to take a HeredocKind is there a reason we still need to pass `heredoc_type`?